smarts/core/agent_interface.py

```diff
@@ -129,6 +129,23 @@ class AgentType(IntEnum):
     """Agent performs trajectory tracking using model predictive control."""
 
 
+@dataclass(frozen=True)
+class AgentsAliveDoneCriteria:
+    total_ego_agents_alive: Optional[int] = None
+    """if set, trigggers the agent to be done if total number of alive ego agents falls below total_agents_alive"""
+    total_social_agents_alive: Optional[int] = None
+    """if set, trigggers the agent to be done if total number of alive social agents falls below total_social_agents_alive"""
+    agent_lists_alive: Optional[List[Tuple[List[str], int]]] = None
+    """a list of agent lists to check alive or not, structure is List[Tuple[list_of_agent_ids, minimum_number_of_agents_alive in the list]]
+    if set, triggers the agent to be done if any list of agents fails to meet its specified minimum_number_of_agents_alive
+    Example: [
+        (['agent1', 'agent2'], 1),
+        (['agent3'], 1)
+    ]
+    Done on the ego agent would be triggered if both agent1 and agent2 is done or agent3 is done
```

Comment:
```suggestion
    This agent's done event would be triggered if both 'agent1' and 'agent2' is done *or* 'agent3' is done.
```

Reply:
is putting `*` around `or` done on purpose (`*or*`)? If so, I am wondering whats the meaning for this?

Reply:
No need for it, it was just an emphasis.